lib/wallet/wallet.js

```diff
@@ -2968,6 +2969,10 @@ class Wallet extends EventEmitter {
     assert(mtx.verifyInputs(this.wdb.height + 1, this.network),
       'TX failed context check.');
 
+    // Set the HD paths.
```

Comment:
Do you mean to always set the paths unless explicitly saying to not set them?

```javascript
> undefined !== false
true
```

Reply:
This is unexpected behavior; great catch!